### PR TITLE
do not support sorting a Dict

### LINKED ISSUE #25: Type piracy?
We're having an issue using Julia from the Juno REPL (which imports `OrderedCollections`), in which `sort()` of a `Dict` returns an `OrderedDict` even though the user didn't import `OrderedCollections`.  The unexpected behavior is described here:

https://discourse.julialang.org/t/juno-repl-behaves-differently-than-terminal-repl/29469/2

The culprit appears to be :
```julia
julia> d = Dict(1 => "a", 3 => "b", 0 => "c")
Dict{Int64,String} with 3 entries:
  0 => "c"
  3 => "b"
  1 => "a"

julia> sort(d)
OrderedCollections.OrderedDict{Int64,String} with 3 entries:
  0 => "c"
  1 => "a"
  3 => "b"

julia> @which sort(d)
sort(d::Dict) in OrderedCollections at /Users/dpo/.julia/packages/OrderedCollections/E21Rb/src/dict_sorting.jl:21
```
Thanks.

### PR DESCRIPTION
**Pull request by dpo**
fixes #25

### CODE REVIEW

**Comment by dpo** [comment]
I wasn't sure if you would rather remove the sort tests altogether.

**Comment by oxinabox** [comment]
Seems good to me.
Let's change the version in the Project.toml to `2.0.0-DEV`
and then we can merge Theis, and check if anything else is pending before we do a release.

**Comment by timholy** [comment]
Yes, this is quite breaking so going to 2.0 seems like a good idea.

**Comment by dpo** [comment]
There's no Project.toml though ?!

**Comment by timholy** [comment]
Definitely want to keep the sort tests, you did the right thing here.

**Comment by timholy** [comment]
You know about `gen_project.jl`, right?

**Comment by dpo** [comment]
Sure do. Wasn't sure you wanted that in the same PR.

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name = "OrderedCollections"
+uuid = "bac558e1-5e72-5ebc-8fee-abe8a469f55d"
+version = "2.0.0-DEV"
+
+[compat]
+julia = "≥ 0.7.0"
```
Comment (timholy) [inline Project.toml]:
```suggestion
julia = "0.7, 1"
```
Will likely break with Julia 2.0 so might as well prevent problems now.

Comment (dpo) [reply Project.toml]: Please go ahead and change what you want. I've no idea about Julia 2.0 !

Comment (timholy) [reply Project.toml]: Just click the button. I don't have permission to modify your repo.

**Comment by timholy** [comment]
If I recall correctly something like 70% of the Julia ecosystem depends on this package. I kind of think we shouldn't tag (or perhaps even merge) this without some testing of the consequences. Any thoughts on how to do that?

File: src/dict_sorting.jl [hunk 2]
```diff
@@ -18,7 +18,6 @@ function sort!(d::OrderedDict; byvalue::Bool=false, args...)
 end
 
 sort(d::OrderedDict; args...) = sort!(copy(d); args...)
-sort(d::Dict; args...) = sort!(OrderedDict(d); args...)
```
Comment (timholy) [inline src/dict_sorting.jl]: Come to think of it, given how widely used this package is, perhaps we should change this to
```julia
@deprecate sort(d::Dict; args...) sort!(OrderedDict(d); args...)
```
(GitHub doesn't let me make mergeable suggestions for deleted lines.)

**Comment by oxinabox** [comment]
SemVer says adding a deprecation is to be in a minor release.
So `1.1`
Followed by removal in `2.0`

**Comment by timholy** [comment]
There's already a 1.1, would have to be 1.2.

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name = "OrderedCollections"
+uuid = "bac558e1-5e72-5ebc-8fee-abe8a469f55d"
+version = "2.0.0-DEV"
```
Comment (ararslan) [inline Project.toml]:
```suggestion
version = "1.2.0"
```

Comment (oxinabox) [reply Project.toml]: ✔️  Assuming we also add the suggestion of deprecate,

File: src/dict_sorting.jl (hunk 2 above)
Comment (dpo) [reply src/dict_sorting.jl]: I had to do this using the Github editor. Hope it's ok.

File: src/dict_sorting.jl [hunk 4]
```diff
@@ -18,7 +18,7 @@ function sort!(d::OrderedDict; byvalue::Bool=false, args...)
...
+@deprecate sort(d::Dict; args...) = sort!(OrderedDict(d); args...)
```
Comment (ararslan) [inline src/dict_sorting.jl]:
```suggestion
@deprecate sort(d::Dict; args...) sort!(OrderedDict(d); args...)
```

**Comment by timholy** [comment]
OK, let's try it. I'll leave it on master for a couple of days and see what problems crop up before we tag a new release.

**Comment by timholy** [comment]
Thanks @dpo!

**Comment by dpo** [comment]
Thank you! One fewer crises in the classroom!

**Pull request merged**